modules/mesh-task/validation.tf

```diff
@@ -3,4 +3,5 @@ locals {
   require_ca_cert_if_tls_enabled             = (var.tls && var.consul_server_ca_cert_arn == "") ? file("ERROR: consul_server_ca_cert_arn must be set if tls is true") : null
   require_client_token_if_acls_enabled       = (var.acls && var.consul_client_token_secret_arn == "") ? file("ERROR: consul_client_token_secret_arn must be set if acls is true") : null
   require_secret_name_prefix_if_acls_enabled = (var.acls && var.acl_secret_name_prefix == "") ? file("ERROR: acl_secret_name_prefix must be set if acls is true") : null
+  require_namespace_if_partition_is_set      = (var.consul_partition != "" && var.consul_namespace == "") ? file("ERROR: consul_namespace must be set if consul_partition is set") : null
```

Comment:
or are we using the fact that it's not an empty string to know that it's consul ent?

Reply:
because we need to support OSS as well as enterprise and you can't use any partition/namespace values with OSS as it causes the Consul agent to return errors.

Reply:
I was chatting with @erichaberkorn about this today. I think there is a bit more work that we can do in terms of improving the UX around AP/NS support in the OSS/Enterprise cases. There are a few permutations that we could handle more gracefully in terms of defaulting partition/namespace values. I think the priority for now is to get this patch out because without it things are broken.

Reply:
> or are we using the fact that it's not an empty string to know that it's consul ent?

Yes, that's correct.